SavedInstances/Core/Core.lua

```diff
@@ -1344,6 +1352,26 @@ function SI:UpdateToonData()
   t.Arena2v2rating = tonumber(GetPersonalRatedInfo(1), 10) or t.Arena2v2rating
   t.Arena3v3rating = tonumber(GetPersonalRatedInfo(2), 10) or t.Arena3v3rating
   t.RBGrating = tonumber(GetPersonalRatedInfo(4), 10) or t.RBGrating
+
+  t.SpecializationIDs = t.SpecializationIDs or {}
+  t.SpecializationIDs[1] = GetSpecializationInfo(1) or t.SpecializationIDs[1] or nil
+  t.SpecializationIDs[2] = GetSpecializationInfo(2) or t.SpecializationIDs[2] or nil
+  t.SpecializationIDs[3] = GetSpecializationInfo(3) or t.SpecializationIDs[3] or nil
+  t.SpecializationIDs[4] = GetSpecializationInfo(4) or t.SpecializationIDs[4] or nil
+
+  -- Solo Shuffle rating is unique to each specialization
+  t.SoloShuffleRating = t.SoloShuffleRating or {}
+  local currentSpecializationId = GetSpecialization()
+  if (currentSpecializationId==1) then
+    t.SoloShuffleRating[1] = GetPersonalRatedInfo(7) or t.SoloShuffleRating[1]
+  elseif (currentSpecializationId == 2) then
+    t.SoloShuffleRating[2] = GetPersonalRatedInfo(7) or t.SoloShuffleRating[2]
+  elseif (currentSpecializationId == 3) then
+    t.SoloShuffleRating[3] = GetPersonalRatedInfo(7) or t.SoloShuffleRating[3]
+  elseif (currentSpecializationId == 4) then
+    t.SoloShuffleRating[4] = GetPersonalRatedInfo(7) or t.SoloShuffleRating[4]
+  end
```

Comment:
use loop and var index

```lua
for i = 1, GetNumSpecializations() do
  t.SpecializationIDs[i] = GetSpecializationInfo(i) or t.SpecializationIDs[i]
end
```

```lua
local currentSpecID = GetSpecialization()
if currentSpecID then
  t.SoloShuffleRating[currentSpecID] = GetPersonalRatedInfo(7) or t.SoloShuffleRating[currentSpecID]
end
```

Reply:
Great suggestions, I didn't know about `GetNumSpecializations()`. Updated.